packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditReducers.ts

```diff
@@ -105,50 +122,80 @@ export function applySynchronizationOp(state: DDSFuzzTestState<SharedTreeFactory
 }
 
 // TODO: Update this function to be done in a more ergonomic way using libraries
-export function generateLeafNodeSchemas(nodeTypes: string[]) {
-	const builder = new SchemaBuilderInternal({ scope: "com.fluidframework.leaf" });
+export function generateLeafNodeSchemas(nodeTypes: string[]): TreeNodeSchema[] {
+	const builder = new SchemaFactory("treeFuzz");
 	const leafNodeSchemas = [];
 	for (const nodeType of nodeTypes) {
 		if (
-			nodeType !== "treefuzz.node" &&
-			nodeType !== "com.fluidframework.leaf.number" &&
-			nodeType !== "com.fluidframework.leaf.string"
+			nodeType !== "treeFuzz.node" &&
+			nodeType !== "treeFuzz.FuzzStringNode" &&
+			nodeType !== "treeFuzz.FuzzNumberNode"
 		) {
-			if (!nodeType.startsWith("com.fluidframework.leaf")) {
-				leafNodeSchemas.push(builder.leaf(nodeType, ValueSchema.Number));
+			const fuzzNodeTypePrefix = "treeFuzz.";
+			if (!nodeType.startsWith(fuzzNodeTypePrefix)) {
+				class GuidNode extends builder.object(nodeType.slice(fuzzNodeTypePrefix.length), {
+					value: builder.required(builder.string),
+				}) {}
+				leafNodeSchemas.push(GuidNode);
 			}
 		}
 	}
-	const library = builder.intoLibrary();
-	return { leafNodeSchemas, library };
+	return leafNodeSchemas;
 }
 export function applySchemaOp(state: FuzzTestState, operation: SchemaChange) {
 	const nodeTypes = getAllowableNodeTypes(state);
-	nodeTypes.push(brand(operation.operation.type));
-	const { leafNodeSchemas, library } = generateLeafNodeSchemas(nodeTypes);
-	const newSchema = createTreeViewSchema(leafNodeSchemas, library);
+	nodeTypes.push(operation.contents.type);
+	const leafNodeSchemas = generateLeafNodeSchemas(nodeTypes);
+	const newSchema = createTreeViewSchema(leafNodeSchemas);
+
+	// Because we need the view for a schema change, and we can only have one view at a time,
+	// we must dispose of the client's view early.
 	const view = viewFromState(state, state.client);
-	view.checkout.updateSchema(intoStoredSchema(newSchema));
+	view.dispose();
+	state.transactionViews?.delete(state.client.channel);
```

Comment:
do we not need to put the new view in the map? (I'm not sure how this map is used, feel free to ignore)

Reply:
Updated to put the new view in the map

Reply:
I'm surprised nothing broke when this was the case. Why is that?

Reply:
It was because the schemaOps were disabled for the tests (as there is currently the double dispose error we run into), and when enabled, the test fails as soon as we call `upgradeSchema`. However, I forgot that the `viewFromState` function checks the map for a cached view, and otherwise creates a new view to return anyways. I am just going to double check this to see if we actually need to store this newView in the map

Reply:
I wasn't able to check in the fuzz tests directly (as it fails as soon as we call `upgradeSchema` for known view disposal issues), but re-created a similar scenario in a test and checked that the tree's storedSchema does get updated once we call `upgradeSchema` on its view.

Since the `viewFromState` function will recreate a new view (getting a simple tree schema from the tree's storedSchema), it should work correctly even if we don't store this newView in the map. That being said, it would also be fine if we set the view in the map as well.

Did you want me to remove this bit of code?

Reply:
I'll leave it up to you :)

Reply:
Sounds good Taylor :)